### PR TITLE
Add cancel sale method

### PR DESCRIPTION
**Pull request by no-donut-for-you**

### CODE REVIEW

**Comment by antoniofilho** [comment]
You missed the views

**Comment by no-donut-for-you** [comment]
Oh :hankey:!

File: lib/fake_braspag/views/sales_cancel.jbuilder [hunk 1]
```diff
@@ -0,0 +1,10 @@
+json.Status 10
+json.ReasonCode 0
+json.ReasonMessage "Successful"
+json.ProviderReturnCode "9"
+json.ProviderReturnMessage "Operation Successful"
+json.Links do |link|
```
Comment (antoniofilho) [inline lib/fake_braspag/views/sales_cancel.jbuilder]: Links can return an empty array

File: lib/fake_braspag/views/sales_cancel_failure.jbuilder [hunk 2]
```diff
@@ -0,0 +1,2 @@
+json.Code 114
```
Comment (antoniofilho) [inline lib/fake_braspag/views/sales_cancel_failure.jbuilder]: The errors should be returned in an array

File: spec/sales_spec.rb [hunk 3]
```diff
@@ -152,4 +150,31 @@
...
   end
+
+  describe 'sale cancelation' do
+    context 'success' do
+      before { ResponseToggler.enable('sale_cancelation') }
+
+      it 'responds with a success response' do
+        put "/v2/sales/2014111703/void"
+
+        expect(last_response).to be_ok
```
Comment (antoniofilho) [inline spec/sales_spec.rb]: You could test the response body too. Wdyt?

**Comment by antoniofilho** [comment]
:+1:

File: lib/fake_braspag/views/sales_cancel_failure.jbuilder [hunk 4]
```diff
@@ -0,0 +1 @@
+json.array! [OpenStruct.new(Code: 114, Message: "Error")], :Code, :Message
```
Comment (bvicenzo) [inline lib/fake_braspag/views/sales_cancel_failure.jbuilder]: what does mean the code 114?

File: lib/fake_braspag/views/sales_cancel.jbuilder [hunk 5]
```diff
@@ -0,0 +1,6 @@
+json.Status 10
```
Comment (bvicenzo) [inline lib/fake_braspag/views/sales_cancel.jbuilder]: what does mean the status code 10, reasonCode 0 etc e etc?

**Comment by bvicenzo** [comment]
:candy: :8ball:

File: lib/fake_braspag/views/sales_cancel_failure.jbuilder (hunk 4 above)
Comment (antoniofilho) [inline lib/fake_braspag/views/sales_cancel_failure.jbuilder]: In this case nothing... is only a fake error response, the code could be anything...

File: lib/fake_braspag/views/sales_cancel.jbuilder (hunk 5 above)
Comment (antoniofilho) [inline lib/fake_braspag/views/sales_cancel.jbuilder]: Voided with successful, this template is a mock from a successful cancellation response.

**Pull request merged**